File: src/org/idea/plugin/pofgen/SerializableField.java
package org.idea.plugin.pofgen;

import com.intellij.psi.PsiField;
import com.intellij.psi.PsiMethod;
import com.intellij.psi.util.PropertyUtil;

/**
 * @author sigito
 */
public class SerializableField {
    private final PsiField psiField;
    private final PsiMethod getter;
    private final PsiMethod setter;

    private final String indexName;
    private final int index;

    public SerializableField(PsiField psiField, int index) {
        this.psiField = psiField;
        this.getter = PropertyUtil.findGetterForField(psiField);
        this.setter = PropertyUtil.findSetterForField(psiField);

        this.indexName = psiField.getName().toUpperCase();
        this.index = index;
    }

    public PsiField getPsiField() {
        return psiField;
    }

    public PsiMethod getGetter() {
        return getter;
    }

    public PsiMethod getSetter() {
        return setter;
    }

    public String getIndexName() {
        return indexName;
    }

    public int getIndex() {
        return index;
    }
}

File: src/org/idea/plugin/pofgen/SerializableField.scala
package org.idea.plugin.pofgen

import com.intellij.psi.codeStyle.{VariableKind, JavaCodeStyleManager}
import com.intellij.psi.util.PropertyUtil
import com.intellij.psi.{PsiMethod, PsiField}

/**
 * @author sigito
 */
class SerializableField(val psiField: PsiField, val index: Int) {
  val typeName = psiField.getType.getCanonicalText
  val name: String = psiField.getName
  val getter: PsiMethod = PropertyUtil.findGetterForField(psiField)
  val setter: PsiMethod = PropertyUtil.findSetterForField(psiField)
  val indexName: String = {
    val codeStyleManager = JavaCodeStyleManager.getInstance(psiField.getProject)
    val suggestedNames = codeStyleManager.suggestVariableName(VariableKind.STATIC_FINAL_FIELD, name, null, null).names

    if (!suggestedNames.isEmpty) suggestedNames(0)
    else name.toUpperCase
  }
}